Update everest-core to 2025.9.0, additional kirkstone backports and fixes

hikinggrass opened the pull request:
Add some additional backports that will soon be used by everest-core:

asyncapi 2.7.1
ftxui 6.1.9
sdbus-c++ 1.6.0
tpm2-openssl 1.2
nodejs 18

Fixes for nativesdk in evcli and pip-stringcase recipes

hikinggrass committed:
Add nativesdk to evcli and pip-stringcase recipes

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Add additional backports:

asyncapi 2.7.1
ftxui 5.0.0
nanopb 0.4.8
sdbus-c++ 1.6.0
tpm2-openssl 1.2

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Update kirkstone to EVerest 2025.8.0

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Merge remote-tracking branch 'origin/feature/update-kirkstone-to-everest-2025.8.0' into feature/additional-kirkstone-backports-and-fixes

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Merge remote-tracking branch 'origin/kirkstone' into feature/additional-kirkstone-backports-and-fixes

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

james-ctc commented on recipes-backports/nanopb/nanopb.bb:
I'm not sure that nanopb is needed anymore.
I think where it is used the generated files have been included rather than running nanopb as part of cmake. Possibly only the runtime is needed
(removed)

james-ctc committed:
feat: added classes for generating a version file
feat: use latest main + patch to test against latest included modules in everest-core
fix: bump libiso15118 and libevse-security to match everest-core
feat: add nodejs-18
feat: default mosquitto to use UNIX socket

Signed-off-by: James Chapman <james.chapman@pionix.de>

james-ctc committed:
feat: automatic TPM2 support (via DISTRO_FEATURES)
fix: removed nonopb - no longer needed

Signed-off-by: James Chapman <james.chapman@pionix.de>

james-ctc committed:
fix: updates to support building custom modules (such as Pionix Cloud)
fix: removed obsolite patch

Signed-off-by: James Chapman <james.chapman@pionix.de>

james-ctc committed:
fix: using everest-core branch main

Signed-off-by: James Chapman <james.chapman@pionix.de>

hikinggrass committed:
Update everest-core to recent main

everest-framework to 0.23.1
libevse-security to 0.9.9
libocpp to 0.30.2
josev to 2025.9.0

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

james-ctc reviewed (approved):
looks good to me

hikinggrass committed:
Update everest-core to 2025.9.0

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Remove tryboot file from everest-core SRC_URI

This was apparently accidentally added

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Add missing ftxui dependency to everest-core

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Bump ftxui dependency to 6.1.9

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

hikinggrass committed:
Remove mosquitto config that was added by mistake

Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

james-ctc reviewed (approved):
looks good to me
can't comment on nodejs - I assume there were no changes from the original layer.
